src/Microsoft.NetCore.Analyzers/Core/Tasks/DoNotCallBeginInvokeOnDelegate.cs

```diff
@@ -0,0 +1,66 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Tasks
+{
+    /// <summary>
+    /// CA1069: Do not call BeginInvoke on a delegate in .NET Core and .NET Standard.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotCallBeginInvokeOnDelegate : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1069";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotCallBeginInvokeOnDelegateTitle), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotCallBeginInvokeOnDelegateMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.DoNotCallBeginInvokeOnDelegateDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Design,
+                                                                             DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                             isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: null,
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                var isNetStandardAssembly = compilationContext.Compilation.ReferencedAssemblyNames.Any(identity => string.Equals(identity.Name, "netstandard", StringComparison.OrdinalIgnoreCase));
```

Comment:
❕ There are two problems with this:

1. It covers .NET Standard 2.x, but misses .NET Standard 1.x
2. It fails to account for .NET Framework 4.7.2 projects that pick up the netstandard facade by referencing a .NET Standard assembly

I believe a better approach here is to exclude the known .NET Framework cases, which are builds that reference one of the following:

* mscorlib, Version=4.0.0.0, PublicKeyToken=b77a5c561934e089
* mscorlib, Version=2.0.0.0, PublicKeyToken=b77a5c561934e089

Reply:
Please add a shared helper method for whatever functionality is implemented. I believe we added similar logic in another analyzer, that would also need to be fixed.

Also adding @stephentoub for suggestions on how to detect this in the best possible way.

Reply:
I extracted detection of compilation target to `CompilationExtensions.DoesTargetDotNetFramework`. However, implementation is 100% valid. The problem is that tests compile targeting .NET Framework, so there would be no way to verify this analyzer. Unless, tests are modified to compile to .NET Standard or .NET Core.

Reply:
➡️ Detection has been updated. Reference assemblies used in testing is a separate consideration moved below.